File: Spring_OAuth2/securityserver/src/main/kotlin/com/example/securityserver/model/domain/user/Authority.java
package com.example.securityserver.model.domain.user;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name="AUTHORITY")
public class Authority implements Serializable {

    private static final long serialVersionUID = 6064399222664228433L;

    @Id
    @Column(name = "AUTHORITY_CODE")
    private int AuthorityCode;

    @Column(name = "AUTHORITY_NAME", nullable = false, unique = true)
    private String authorityName;

    @Column(name = "AUTHORITY_DESC")
    private String authorityDesc;

    public Authority() {

    }

    public int getAuthorityCode() {
        return AuthorityCode;
    }

    public void setAuthorityCode(int authorityCode) {
        AuthorityCode = authorityCode;
    }

    public String getAuthorityName() {
        return authorityName;
    }

    public void setAuthorityName(String authorityName) {
        this.authorityName = authorityName;
    }

    public String getAuthorityDesc() {
        return authorityDesc;
    }

    public void setAuthorityDesc(String authorityDesc) {
        this.authorityDesc = authorityDesc;
    }

    public Authority(int authorityCode, String authorityName, String authorityDesc) {
        AuthorityCode = authorityCode;
        this.authorityName = authorityName;
        this.authorityDesc = authorityDesc;
    }

    @Override
    public String toString() {
        return "Authority{" +
                "AuthorityCode=" + AuthorityCode +
                ", authorityName='" + authorityName + '\'' +
                ", authorityDesc='" + authorityDesc + '\'' +
                '}';
    }
}

File: Spring_OAuth2/securityserver/src/main/kotlin/com/example/securityserver/model/domain/user/Authority.kt
package com.example.securityserver.model.domain.user;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

import lombok.Getter;
import lombok.Setter;
@Entity
@Table(name = "AUTHORITY")
class Authority : Serializable {
    @Id
    var authorityCode: Int? = null

    var authorityName: String? = null

    var authorityDesc: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Authority

        return authorityCode == other.authorityCode
    }

    override fun hashCode(): Int {
        return authorityCode?.hashCode() ?: 0
    }

    override fun toString(): String {
        return "Authority(authorityCode=$authorityCode, authorityName=$authorityName, authorityDesc=$authorityDesc)"
    }
}

